packages/firestore/src/util/misc.ts

```diff
@@ -121,3 +121,12 @@ export function immediateSuccessor(s: string): string {
   // Return the input string, with an additional NUL byte appended.
   return s + '\0';
 }
+
+/**
+ * Detects whether we are running in a React Native app.
+ */
+export function isReactNative(): boolean {
```

Comment:
Could we put this in a platform-detection module?

Maybe even combined with our browser version detection stuff?